platform/feature-api/api/pod-api/src/api.ts

```diff
@@ -199,6 +182,31 @@ export interface PolyLifecycle {
     startFeature(id: string, background: boolean): Promise<void>;
 }
 
+/**
+ * `Endpoint` is the API features communicate with in order to perform fetch requests
+ */
+export interface Endpoint {
+    /**
+     * Perform a http post request via the endpoint in the pod
+     * @param body the necessary content of the call
+     * @returns an empty response or throws an error if something goes wrong
+     * @throws if an unsupported request goes through, if an endpoint is not reached or if a user denies a request
+     */
+    send(
+        endpointId: string,
+        payload: string,
+        contentType?: string,
+        authToken?: string
+    ): Promise<void>;
+
+    /**
+     * Perform a http get request via the endpoint in the pod
+     * @returns a promise with the response
+     * @throws if an unsupported request goes through, if an endpoint is not reached, if a user denies a request or if response is null
+     */
+    get(endpointId: string, contentType?: string, authToken?: string): Promise<string>;
```

Comment:
Again, endpointID as well as possibly authToken might be instance variables. Even contentType. The endpoint should know what kind of types it supports (although it's likely that it supports more than one).